src/support/lock_ext.c

```diff
@@ -0,0 +1,86 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __wt_ext_spin_init --
+ *     Allocate and initialize a spinlock.
+ */
+int
+__wt_ext_spin_init(WT_EXTENSION_API *wt_api, WT_SESSION *session,
+  WT_EXTENSION_SPINLOCK *ext_spinlock, const char *name)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *default_session;
+    WT_SPINLOCK *lock;
+
+    WT_UNUSED(session); /*Unused parameters */
+    ext_spinlock->spinlock = NULL;
+    default_session = ((WT_CONNECTION_IMPL *)wt_api->conn)->default_session;
```

Comment:
Can you add a comment to explain why we need to use the `default_session` here instead of the passed `session`?

Reply:
Could you remove the WT_SESSION parameter? There are other extension APIs that don't take a session handle.